### PR TITLE
Replace old trainer with the new one.

### PR DESCRIPTION
**Pull request by rahul1980**
- Removes third_party/syntaxnet and its usage.
- Switches train.sh to use the new trainer.
- Cleans up the new link feature implementation to always return absolute instead of relative indices.
- Modifies README.md to reflect these changes. 
- Modifies WORKSPACE to reflect changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,15 +59,15 @@ get all the submodules.
...
+so they need to be installed. Additionally,
 SLING uses [Bazel](https://bazel.build/) as the build system, so you need to
 install Bazel in order to build the SLING parser.
 
 ```shell
-sudo pip install -U protobuf==3.4.0
-sudo pip install https://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-1.4.0-cp27-none-linux_x86_64.whl
+# Change to your favorite version as needed.
+sudo pip install http://download.pytorch.org/whl/cpu/torch-0.3.1-cp27-cp27mu-linux_x86_64.whl 
+sudo pip install torchvision
```
Comment (ringgaard) [inline README.md]: Is torchvision strictly needed, and if so, what is it used for?

File: sling/nlp/parser/trainer/pytorch_modules.py [hunk 2]
```diff
@@ -367,12 +367,13 @@ def _ff_output(
       assert len(indices) == f.num
 
       for index in indices:
+        #print f.name, index
```
Comment (ringgaard) [inline sling/nlp/parser/trainer/pytorch_modules.py]: You can probably remove this debug comment.

Comment (rahul1980) [reply sling/nlp/parser/trainer/pytorch_modules.py]: Done.

File: README.md (hunk 1 above)
Comment (rahul1980) [reply README.md]: You are right, not needed. This is a package for computer vision/image processing.

**Pull request merged**